File: backend/models/FamilyModel/Family.cs

using FamilyPlanner.Models.PlannedDayModel;
using FamilyPlanner.Models.UserModel;

namespace FamilyPlanner.Models.FamilyModel {
    public sealed class Family {
        public int Id { get; set; }
        public string Name { get; set; }
        public string? UserUid { get; set; }
        public string? FamilyColor { get; set; }
        public List<User>? Members { get; set; }
        public List<PlannedDay>? PlannedDays { get; set; }
    }
}

File: familyPlanner/src/models/family.ts
import { PlannedDay } from "./plannedDay";
import { User } from "./user";
import { WeeklyPlanner } from "./weeklyPlanner";

export interface Family {
    id: number;
    name: string;
    userUid?: string;
    familyColor?: string;
    members?: User[];
    plannedDays?: PlannedDay[];
}

export interface FamilyInput {
    name: string;
    members?: User[];
    plannedDays?: PlannedDay[];
}